src/crc32.rs

```diff
@@ -21,6 +23,15 @@ impl Crc32 {
     pub fn new(crc: CRC) -> Self {
         let mut new = Self { periph: crc };
         new.init();
+
+        unsafe {
+            // NOTE(unsafe) this reference will only be used for atomic writes with no side effects.
+            let rcc_raw = unsafe { &(*RCC::ptr()) };
```

Comment:
This is already inside a unsafe block, so the second unsafe is unnecessary and emits a warning.

Reply:
Corrected.